### ISSUE TITLE
client side copypac leaves extra files if target package already exists

### ISSUE THREAD

**Issue by tuminoid** [openSUSE/osc#181]
Say you have `package1`, having `file1` and `file2`. 
You then `osc copypac prj package1 prj package2`. Now `package2` has `file1` and `file2`.

If you then delete `file2` from `package1`, and `osc copypac -c prj package1 prj package2`, `package2` will still have both files.

If you `osc copypac prj package1 prj package2` again, now `package2` will have only `file1`, which is correct.

Thus, client side copy results in very different thing versus server side copy. This is especially bad when you copy a package from another scheduler, as it is always client-side copy, and you end up with extra files.

**Comment by marcus-h** [comment]
Fixed in commit ffb37960ffdc0e019344898cb62375cb5b237161. Thanks for
spotting this!

**Comment by tuminoid** [comment]
Hi @marcus-h. 

Unfortunately this implementation has even more severe bug. Adding a new file to `pkg1` and client-side copying over `pkg2`, results in all the old files in `pkg2` being deleted, leaving only new file.

```
osc co <prj> && cd <prj>
osc mkpac source && cd source
touch file1 && osc add file1 && osc commit file1 -m "file1"
osc copypac -c <prj> source <prj> target
# both will have file 1
touch file2 && osc add file2 && osc commit file2 -m "file2"
osc copypac -c <prj> source <prj> target
# target will have only file2, source has file1 and file2
```

You can see this error in osc output, not printing file1 for copying:
```
source$ osc copypac -c home:tuomotanskanen:osc_test source home:tuomotanskanen:osc_test target
Copying files...
  file2
Done.
```

**Comment by marcus-h** [comment]
On 2016-02-12 01:02:54 -0800, Tuomo Tanskanen wrote:
> Unfortunately this implementation has even more severe bug. Adding a new file to `pkg1` and client-side copying over `pkg2`, results in all the old files in `pkg2` being deleted, leaving only new file.
> 
Oops... fixed in commit 6a69ecc137a51bd243268f8f2d53ce68e8a86d69.

> You can see this error in osc output, not printing file1 for copying:
> ```
> source$ osc copypac -c home:tuomotanskanen:osc_test source home:tuomotanskanen:osc_test target
> Copying files...
>   file2
> Done.
> ```
That's actually a "feature", because it (technically) only transfers file2
(file1 is already present). But yes... from a users perspective it might be
confusing. Should it print all files?

**Comment by tuminoid** [comment]
Thanks for the quick turnaround. :+1: 

Well if you think it like that, I guess it is OK to print just the transferred files. I just personally assumed it copied all the files anyways. I'm OK as long as the end result is equivalent of server-side copypac.
